File: TransactionRegister/Controller/Reusable/TXViewController.m
//
//  TXViewController.m
//  TransactionRegister
//
//  Created by Eric Romrell on 10/26/16.
//  Copyright © 2016 Eric Romrell. All rights reserved.
//

#import "TXViewController.h"

@implementation TXViewController

-(void)showError:(TXError *)error {
	[self showError:error withCallback:^(UIAlertAction *alert) {
		[self.navigationController popViewControllerAnimated:YES];
	}];
}

-(void)showError:(TXError *)error withCallback:(void (^)(UIAlertAction *))callback {
	UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error" message:error.readableMessage preferredStyle:UIAlertControllerStyleAlert];
	[alert addAction:[UIAlertAction actionWithTitle:@"Dismiss" style:UIAlertActionStyleCancel handler:callback]];
	[self presentViewController:alert animated:YES completion:nil];
}

@end

File: TransactionRegister/Controller/Reusable/TXViewController.swift
//
//  TXViewController.swift
//  TransactionRegister
//
//  Created by Eric Romrell on 1/20/17.
//  Copyright © 2017 Eric Romrell. All rights reserved.
//

import UIKit

class TXViewController: UIViewController {
	func showError(error: TXError) {
		showError(error: error, callback:{ (_) in
			_ = self.navigationController?.popViewController(animated: true)
		})
	}
	
	func showError(error: TXError, callback:@escaping (UIAlertAction) -> Void) {
		let alert = UIAlertController(title: "Error", message: error.readableMessage, preferredStyle: .alert)
		
		
		alert.addAction(UIAlertAction(title: "Dismiss", style: .cancel, handler: callback))
		self.present(alert, animated: true, completion: nil)
	}
}
